tools/create_repository.py

```diff
@@ -335,10 +335,7 @@ def create_repository(
         info_file.write(info_contents)
 
     # Calculate the signature.
-    digest = hashlib.md5(info_contents).hexdigest()
-    with open(checksum_path, 'w') as sig:
-        sig.write(digest)
-
+    generate_checksum(info_path)
```

Comment:
I think the docs say that the checksum should be for the raw addon.xml contents, not the zipped addon.xml.gz contents. That means that info_contents should be used as the input to hashlib, not the info_path file.